crates/net/src/http/response.rs

```diff
@@ -0,0 +1,281 @@
+    ///
+    /// # Note
+    ///
+    /// This method also sets the `Content-Type` header to `application/json`
+    #[cfg(feature = "json")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "json")))]
+    pub fn json<T: serde::Serialize + ?Sized>(self, value: &T) -> Result<Response, Error> {
+        let json = serde_json::to_string(value)?;
+        self.header("Content-Type", "application/json")
+            .body(Some(json.as_str()))
+    }
+
+    /// Set the response body and return the response
+    pub fn body<T>(mut self, data: T) -> Result<Response, Error>
+    where
+        T: IntoRawResponse,
+    {
+        self.options.headers(&self.headers.into_raw());
+        let init = self.options;
+
+        data.into_raw(init).map(Response).map_err(js_to_error)
+    }
+}
+
+impl IntoRawResponse for Option<&web_sys::Blob> {
+    fn into_raw(self, init: ResponseInit) -> Result<web_sys::Response, JsValue> {
+        web_sys::Response::new_with_opt_blob_and_init(self, &init)
+    }
+}
+
+impl IntoRawResponse for Option<&js_sys::Object> {
+    fn into_raw(self, init: ResponseInit) -> Result<web_sys::Response, JsValue> {
+        web_sys::Response::new_with_opt_buffer_source_and_init(self, &init)
+    }
+}
+
+impl IntoRawResponse for Option<&mut [u8]> {
+    fn into_raw(self, init: ResponseInit) -> Result<web_sys::Response, JsValue> {
+        web_sys::Response::new_with_opt_u8_array_and_init(self, &init)
+    }
+}
+
+impl IntoRawResponse for Option<&web_sys::FormData> {
+    fn into_raw(self, init: ResponseInit) -> Result<web_sys::Response, JsValue> {
+        web_sys::Response::new_with_opt_form_data_and_init(self, &init)
+    }
+}
+
+impl IntoRawResponse for Option<&web_sys::UrlSearchParams> {
+    fn into_raw(self, init: ResponseInit) -> Result<web_sys::Response, JsValue> {
+        web_sys::Response::new_with_opt_url_search_params_and_init(self, &init)
+    }
+}
+
+impl IntoRawResponse for Option<&str> {
+    fn into_raw(self, init: ResponseInit) -> Result<web_sys::Response, JsValue> {
+        web_sys::Response::new_with_opt_str_and_init(self, &init)
+    }
+}
+
+impl IntoRawResponse for Option<&web_sys::ReadableStream> {
+    fn into_raw(self, init: ResponseInit) -> Result<web_sys::Response, JsValue> {
+        web_sys::Response::new_with_opt_readable_stream_and_init(self, &init)
+    }
+}
+
+/// trait which allow consuming self into a raw web_sys::Response
+pub trait IntoRawResponse {
+    /// function which converts self and a web_sys::ResponseInit into a result to a
+    /// web_sys::Response
+    fn into_raw(self, init: ResponseInit) -> Result<web_sys::Response, JsValue>;
+}
+
+impl Default for ResponseBuilder {
+    fn default() -> Self {
+        Self {
+            headers: Headers::new(),
+            options: web_sys::ResponseInit::new(),
+        }
+    }
+}
+
+impl fmt::Debug for ResponseBuilder {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Response Builder")
```

Comment:
```suggestion
        f.debug_struct("ResponseBuilder")
```

The name provided to the debug implementation should match the name of the struct.